server.js

```diff
@@ -47,79 +50,189 @@ function Events(eventData) {
   }
-  response.status(errorObj.status).send(errorObj);
+  response.status(500).send(errorObj);
 }
 
-app.get('/location', (request, response) => {
+// Location data
+function checkNewLocation(request, response) {
   let searchQuery = request.query.data;
   let sqlQuery = `SELECT * FROM locations WHERE search_query='${searchQuery}';`
   client.query(sqlQuery)
     .then(queryResult => {
       if (queryResult.rowCount > 0){
         console.log('In database, sending row 1');
+        currentCity = queryResult.rows[0].search_query;
+        lat = queryResult.rows[0].latitude;
+        lng = queryResult.rows[0].longitude;
         response.send(queryResult.rows[0])
       }
       else {
         let url = `https://maps.googleapis.com/maps/api/geocode/json?address=${searchQuery}&key=${process.env.GEOCODE_API_KEY}`
-        superagent.get(url)
-          .then(superagentResults => {
-            let results = superagentResults.body.results[0];
-            let locations = new Locations(searchQuery, results);
-            //send to database
-            let sql = 'INSERT INTO locations (search_query, formatted_query, latitude, longitude) VALUES ($1, $2, $3, $4);'
-            let values = [searchQuery, locations.formattedQuery, lat, lng];
-            client.query(sql, values)
-              .then(pgResults => {
-                console.log('Asking Google, sending data to database');
-              })
-              .catch(error => errorHandler(error));
-            response.send(locations);
-          })
+        superagentGetLocation(url, searchQuery, response);
       }
       //send to front end
     })
     .catch(err => errorHandler(err, response));
-})
+}
 
-app.get('/weather', (request, response) => {
+function superagentGetLocation(url, searchQuery, response){
+  superagent.get(url)
+    .then(superagentResults => {
+      let locations = newLocation(superagentResults, searchQuery)
+      sendSQLLocation(searchQuery, locations);
+      //send to database
+      response.send(locations);
+    })
+}
+function newLocation(superagentResults, searchQuery){
+  let results = superagentResults.body.results[0];
+  let locations = new Locations(searchQuery, results);
+  return locations;
+}
+
+function sendSQLLocation(searchQuery, locations){
+  let sql = 'INSERT INTO locations (search_query, formatted_query, latitude, longitude) VALUES ($1, $2, $3, $4);'
+  let values = [searchQuery, locations.formattedQuery, lat, lng];
+  client.query(sql, values)
+    .then(pgResults => {
+      console.log('Asking Google, sending data to database');
+    })
+    .catch(error => errorHandler(error));
+}
+
+//Weather Data
+function getWeather(request, response) {
   let searchQuery = request.query.data;
+  console.log(searchQuery);
   let url = `https://api.darksky.net/forecast/${process.env.WEATHER_API_KEY}/${lat},${lng}`
+  superagentWeather(url, searchQuery, response)
+}
+function superagentWeather(url, searchQuery, response){
```

Comment:
I am curious why you put your superagent request into a separate function than your getWeather function above? Does it seem efficient? Do you prefer this? Usually they living inside the same function.